common/js/src/messaging/message-channel-pool.js

```diff
@@ -60,38 +64,73 @@ var MessageChannelPool = GSC.MessageChannelPool;
 
 /**
  * @param {string} extensionId
- * @return {goog.messaging.AbstractChannel}
+ * @return {!Array.<!goog.messaging.AbstractChannel>}
  */
-MessageChannelPool.prototype.getChannel = function(extensionId) {
-  return this.channels_.get(extensionId, null);
+MessageChannelPool.prototype.getChannels = function(extensionId) {
+  return this.channels_.get(extensionId);
+};
+
+/**
+ * Returns the extensionId's of all the connected channels.
+ * @return {!Array.<string>}
+ */
+MessageChannelPool.prototype.getExtensionIds = function() {
+  return this.channels_.getKeys();
 };
 
 /**
- * @param {!goog.messaging.AbstractChannel} messageChannel
  * @param {string} extensionId
+ * @param {!goog.messaging.AbstractChannel} messageChannel
  */
 MessageChannelPool.prototype.addChannel = function(
-    messageChannel, extensionId) {
-  if (this.getChannel(extensionId)) {
+    extensionId, messageChannel) {
+  if (this.channels_.containsEntry(extensionId, messageChannel)) {
     GSC.Logging.failWithLogger(
         this.logger, 'Tried to add a channel that was already present');
   }
   this.logger.fine('Added a new channel, extension id = ' + extensionId);
-  this.channels_.set(extensionId, messageChannel);
+  this.channels_.add(extensionId, messageChannel);
   messageChannel.addOnDisposeCallback(
-      this.handleChannelDisposed_.bind(this, extensionId));
+      this.handleChannelDisposed_.bind(this, extensionId, messageChannel));
+  this.fireOnUpdateListeners_();
+};
+
+/**
+ * @param {function(!Array.<string>)} listener
+ * @param {!Object=} opt_scope
+ */
+MessageChannelPool.prototype.addOnUpdateListener = function(
+    listener, opt_scope) {
+  this.logger.fine('Added an OnUpdateListener');
+  this.onUpdateListeners_.push(
+      goog.isDef(opt_scope) ? goog.bind(listener, opt_scope) : listener);
+  // Fire it once immediately to update.
+  this.fireOnUpdateListeners_();
+};
+
+/**
+ * @private
+ */
+MessageChannelPool.prototype.fireOnUpdateListeners_ = function() {
+  this.logger.fine('Firing onUpdateListeners_');
```

Comment:
nit: The log message here is not very clear, please update it to exclude the name of the private member.
(Sorry, I didn't post this comment for the previous revision.)